lib/core/test/unit/Cardano/Wallet/Api/ServerSpec.hs

```diff
@@ -214,3 +218,60 @@ spec = describe "API Server" $ do
                 HF.EraSummary start (HF.EraEnd end) era1Params
             int = HF.mkInterpreter summary
         in mkTimeInterpreter nullTracer startTime (pure int)
+
+mockNetworkLayer :: NetworkLayer IO a
+mockNetworkLayer = NetworkLayer
+    { nextBlocks = error "nextBlocks: not implemented"
+    , initCursor = error "initCursor: not implemented"
+    , destroyCursor = error "destroyCursor: not implemented"
+    , cursorSlotNo = error "cursorSlotNo: not implemented"
+    , currentNodeEra = error "currentNodeEra: not implemented"
+    , currentNodeTip = error "currentNodeTip: not implemented"
+    , watchNodeTip = error "watchNodeTip: not implemented"
+    , currentProtocolParameters = error "currentProtocolParameters: not implemented"
+    , currentSlottingParameters = error "currentSlottingParameters: not implemented"
+    , postTx = error "postTx: not implemented"
+    , stakeDistribution = error "stakeDistribution: not implemented"
+    , getAccountBalance = error "getAccountBalance: not implemented"
+    , timeInterpreter = error "timeInterpreter: not implemented"
+    }
+
+errorHandlingSpec :: Spec
+errorHandlingSpec = describe "liftHandler and toServerError" $ do
+    let testWalletHandler
+            :: IsServerError e
+            => ExceptT e IO a
+            -> IO (Either ServerError a)
+        testWalletHandler = runHandler . liftHandler
+
+    -- Check that an example error handled by the wallet API server produces a
+    -- structured JSON error message response.
+    it "ErrNoSuchWallet" $ do
+        let handler :: ExceptT ErrNoSuchWallet IO ()
+            handler = throwE $ ErrNoSuchWallet wid
+            wid = unsafeFromText "0000000000000000000000000000000000000000"
+        let expectedErr = ServerError
+                { errHTTPCode = 404
+                , errReasonPhrase = "Not Found"
+                , errBody = mconcat
+                    [ "{\"code\":\"no_such_wallet\",\"message\":\"I couldn't "
+                    , "find a wallet with the given id: 0000000000000000000000000000000000000000\"}" ]
```

Comment:
We could avoid the message here, this is tested in other layers and will only be painful to maintain over time.

Reply:
OK I changed the assertions to be more lenient and maintainable.
As far as I can see, the error message is tested by integration tests but nowhere else.